wagtail/contrib/redirects/tests/test_reports_view.py

```diff
@@ -70,13 +81,27 @@ def test_csv_export(self):
     def test_xlsx_export(self):
         Redirect.add_redirect("/from", "/to", True)
 
-        response = self.get(params={"export": "xlsx"})
+        with self.assertNumQueries(4):
+            response = self.get(params={"export": "xlsx"})
+            workbook_data = response.getvalue()
+
         self.assertEqual(response.status_code, 200)
 
-        workbook_data = response.getvalue()
         worksheet = load_workbook(filename=BytesIO(workbook_data))["Sheet1"]
         cell_array = [[cell.value for cell in row] for row in worksheet.rows]
 
         self.assertEqual(cell_array[0], ["From", "To", "Type", "Site"])
         self.assertEqual(len(cell_array), 2)
         self.assertEqual(cell_array[1], ["/from", "/to", "permanent", None])
+
+    def test_num_queries(self):
+        for i in range(3):
+            Redirect.add_redirect(f"/from{i}", "/to", False)
+            Redirect.add_redirect(f"/from-site{i}", "/to", False, site=self.site)
+            Redirect.add_redirect(f"/to-page{i}", self.page, False)
+
+        with self.assertNumQueries(5):
```

Comment:
suggestion: it would be good to note what the queries are in all of the asserts so we can cross-reference things in the future